### PR TITLE
Test connection with user and password

### PR DESCRIPTION
**Pull request by NicoYuan1986**
Issue: #18592
Signed-off-by: “nico” <Nico_1986@163.com>

### CODE REVIEW

File: tests/python_client/testcases/test_connection.py [hunk 1]
```diff
@@ -974,4 +1072,177 @@ def test_parameters_with_invalid_address_port(self, host, port, connect_name):
...
+        # 1.default user login
+        self.connection_wrap.connect(host=host, port=port, user=ct.default_user,
+                                     password=ct.default_password, check_task=ct.CheckTasks.ccr)
+
+        # 2.create the first credential successfully
+        self.utility_wrap.create_user(user=user, password=ct.default_password)
+
+        # 3.create the second credential with the same user
+        self.utility_wrap.create_user(user=user, password=ct.default_password,
+                                            check_task=ct.CheckTasks.err_res, check_items={ct.err_code: 29})
```
Comment (binbinlv) [inline tests/python_client/testcases/test_connection.py:1126]: Please format this line (alignment)

Comment (binbinlv) [inline tests/python_client/testcases/test_connection.py:1126]: Please format this line (alignment)

File: tests/python_client/testcases/test_connection.py [hunk 2]
```diff
@@ -974,4 +1072,177 @@ def test_parameters_with_invalid_address_port(self, host, port, connect_name):
...
+        """
+        target: test the password when create credential
+        method: make the length of user exceed the limitation [6, 256]
+        excepted: the creation is false
+        """
+        self.connection_wrap.connect(host=host, port=port, user=ct.default_user,
+                                     password=ct.default_password, check_task=ct.CheckTasks.ccr)
+        user = "alice"
+        self.utility_wrap.create_user(user=user, password=password,
+                                            check_task=ct.CheckTasks.err_res, check_items={ct.err_code: 5})
```
Comment (binbinlv) [inline tests/python_client/testcases/test_connection.py:1140]: Please format this line (alignment)

File: tests/python_client/testcases/test_connection.py [hunk 3]
```diff
@@ -974,4 +1072,177 @@ def test_parameters_with_invalid_address_port(self, host, port, connect_name):
...
+    def test_connect_with_invalid_user_connection(self, host, port, user):
+        """
+        target: test the nonexistent to connect
+        method: connect with the nonexistent user
+        excepted: connected is false
+        """
+        self.connection_wrap.connect(host=host, port=port, user=user, password="abc123",
+                                     check_task=ct.CheckTasks.ccr)
+        self.utility_wrap.list_collections(check_task=ct.CheckTasks.err_res,
+                                                 check_items={ct.err_code: 1})
```
Comment (binbinlv) [inline tests/python_client/testcases/test_connection.py:1153]: Please format this line (alignment)

File: tests/python_client/testcases/test_connection.py [hunk 4]
```diff
@@ -974,4 +1072,177 @@ def test_parameters_with_invalid_address_port(self, host, port, connect_name):
...
+    def test_delete_root_when_delete_credential(self, host, port):
+        """
+        target: test deleting user root when deleting credential
+        method: connect and then delete the user root
+        excepted: delete is false
+        """
+        self.connection_wrap.connect(host=host, port=port, user=ct.default_user,
+                                     password=ct.default_password, check_task=ct.CheckTasks.ccr)
+        self.utility_wrap.delete_user(user=ct.default_user, check_task=ct.CheckTasks.err_res,
+                                      check_items={ct.err_code: 31})
```
Comment (binbinlv) [inline tests/python_client/testcases/test_connection.py:1248]: At the end of one file, one more "blank" could be added.

File: tests/python_client/testcases/test_connection.py [hunk 5]
```diff
@@ -845,6 +845,104 @@ def test_parameters_with_address_connection(self, host, port, connect_name):
...
+        user = "nico"
+        password = "wertyu567"
+        self.utility_wrap.create_user(user=user, password=password, check_task=ct.CheckTasks.ccr)
+        self.connection_wrap.connect(host=host, port=port, user=user,
+                                     password=password, check_task=ct.CheckTasks.ccr)
+
+    @pytest.mark.tags(ct.CaseLabel.L3)
+    @pytest.mark.parametrize("old_password", ["abc1234"])
+    @pytest.mark.parametrize("new_password", ["abc12345"])
+    def test_user_old_password_when_reset_password(self, host, port, user, old_password, new_password):
```
[the remark this answers is not in the document]
Comment (NicoYuan1986) [reply tests/python_client/testcases/test_connection.py:881]: I've already made the changes

File: tests/python_client/testcases/test_connection.py [hunk 6]
```diff
@@ -845,6 +845,119 @@ def test_parameters_with_address_connection(self, host, port, connect_name):
...
+        self.connection_wrap.connect(host=host, port=port, user=ct.default_user,
+                                     password=ct.default_password, check_task=ct.CheckTasks.ccr)
+
+        #2. create 2 users
+        self.utility_wrap.create_user(user="user1", password="abc123")
+        self.utility_wrap.create_user(user="user2", password="abc123")
+
+        #3. list all users
+        res = self.utility_wrap.list_usernames()[0]
+        assert "user1" in res
```
Comment (binbinlv) [inline tests/python_client/testcases/test_connection.py:912]: Here this two sentences could be modified as one.

**Pull request merged**